internal/agentseed/agentseed.go

```diff
@@ -0,0 +1,135 @@
+package agentseed
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/google/uuid"
+	"github.com/grafana/agent/pkg/flow/logging/level"
+	"github.com/prometheus/common/version"
+)
+
+// AgentSeed identifies a unique agent
+type AgentSeed struct {
+	UID       string    `json:"UID"`
+	CreatedAt time.Time `json:"created_at"`
+	Version   string    `json:"version"`
+}
+
+const filename = "agent_seed.json"
+
+var dataDir = ""
+var logger log.Logger
+
+var savedSeed *AgentSeed
+
+// Init should be called by an app entrypoint as soon as it can to configure where the unique seed will be stored.
+// dir is the directory where we will read and store agent_seed.json
+// If left empty it will default to $APPDATA or /tmp
+func Init(dir string, log log.Logger) {
+	dataDir = dir
+	logger = log
+}
+
+// Get will return a unique uuid for this agent.
+// Seed will be saved in agent_seed.json
+// If path is not empty, that will be the "preferred" place to read and save it.
+// If it is empty, we will fall back to $APPDATA on windows or /tmp on *nix systems to read the file.
+func Get() (seed *AgentSeed) {
```

Comment:
`Get` can be understood as a side-effect operation, but in this case we initialize and store the seed. How about a more descriptive name? (`Fetch`, `FindOrCreate`, etc)